translate/translate.js

```diff
@@ -18,156 +18,128 @@
 const Translate = require('@google-cloud/translate');
 
 // [START translate_detect_language]
-function detectLanguage (input, callback) {
+function detectLanguage (input) {
   // Instantiates a client
-  const translate = Translate({
-    // The Translate API uses an API key for authentication. This sample looks
-    // for the key in an environment variable.
-    key: process.env.TRANSLATE_API_KEY
-  });
+  const translate = Translate();
 
   // Detects the language. "input" can be a string for detecting the language of
   // a single piece of text, or an array of strings for detecting the languages
   // of multiple texts.
-  translate.detect(input, (err, result) => {
-    if (err) {
-      callback(err);
-      return;
-    }
+  return translate.detect(input)
+    .then((results) => {
+      let detections = results[0];
+
+      if (!Array.isArray(detections)) {
```

Comment:
Nit: I know it's silly, but [there's a library for that](https://www.npmjs.com/package/arrify). :stuck_out_tongue:

(If you decide to use it, please fix this throughout your PR.)